presto-main/src/main/java/com/facebook/presto/execution/PidTaskExecutorController.java

```diff
@@ -0,0 +1,95 @@
+/*
```

Comment:
I think it would be nicer that this commit is split to a few:
 * Introduce TaskExecutorController and StaticTaskExectorController and use it in constructor
 * Adds mechanism of adjusting number of threads but with StaticTaskExectorController
 * Add PidTaskExecutorController